crates/sui-proxy/src/ip.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use ipnetwork::IpNetwork;
+use multiaddr::Multiaddr;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+pub fn to_multiaddr(addr: IpAddr) -> Multiaddr {
+    match addr {
+        IpAddr::V4(a) => Multiaddr::from(a),
+        IpAddr::V6(a) => Multiaddr::from(a),
+    }
+}
+
+/// is_private makes a decent guess at determining of an addr is publicly routable.
+pub fn is_private(addr: IpAddr) -> bool {
+    match addr {
+        IpAddr::V4(a) => is_private_v4(a),
+        IpAddr::V6(a) => is_private_v6(a),
+    }
+}
+
+/// is_private_v4 will say just that, is it private? we ignore 169.254.0.0/16 in this consideration
+fn is_private_v4(addr: Ipv4Addr) -> bool {
+    // special case we will allow
+    let allowed_private: IpNetwork = "169.254.0.0/16".parse().unwrap();
+    if allowed_private.contains(IpAddr::V4(addr)) {
+        // intentional
+        return false;
+    }
+    addr.is_private()
+}
+
+/// is_private_v6 and the funcs below are based on an unstable const fn in core. yoinked it.
+/// taken from https://doc.rust-lang.org/nightly/src/core/net/ip_addr.rs.html#1501-1531
```

Comment:
I doubt this code this url is pointing at is going to remain constant, doesn't the code in nightly change frequently?

Reply:
updated. good point.